plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -99,8 +100,25 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
     // Lambda loads user function using an absolute path.
     let filename = path.resolve(taskRoot, moduleRoot, module);
     if (!filename.endsWith('.js')) {
```

Comment:
I wonder, could we do a quick optimization and also check if the filename ends with`.cjs` here? In that case we can also just carry on without needing to do all the expensive operations below.

Reply:
I ran a test and we might be able to do something with fs.stat here

```
import fs from 'node:fs/promises';

console.time('stat');
await fs.stat('./test.js').catch(e=>{});
console.timeEnd('stat')


console.time('listdir');
await fs.readdir('./ping');
console.timeEnd('listdir')
```

```
stat: 0.111ms
listdir: 0.416ms
```
Obviously that performance hit is pretty significant but I could check for a .js file then fallback to .cjs if its not found.

I suspect this will also need some work once the otel.js library supports esm

Reply:
Yeah I like this approach the best, it's simple, not perfect, but an improvement over what we have today. I'd be good with this change.

Reply:
Just realized the numbers are .11ms not 111ms :sweat_smile: